LEGO1/mxdsfile.cpp

```diff
@@ -1,6 +1,139 @@
 #include "mxdsfile.h"
 
+#include "stdio.h"
+
+#define SI_MAJOR_VERSION 2
+#define SI_MINOR_VERSION 2
+
+// OFFSET: LEGO1 0x100bffd0
+void MxDSSource::SomethingWhichCallsRead(void* pUnknownObject)
+{
+  // TODO: Calls read, reading into a buffer somewhere in pUnknownObject.
+  Read(NULL, 0);
+}
+
+// OFFSET: LEGO1 0x100bfff0
+long MxDSSource::GetLengthInDWords()
+{
+  return m_lengthInDWords;
+}
+
+// OFFSET: LEGO1 0x100cc4b0
+MxDSFile::MxDSFile(const char *filename, unsigned long skipReadingChunks)
+{
+  m_filename = filename;
+  m_skipReadingChunks = skipReadingChunks;
+}
+
+// OFFSET: LEGO1 0x100bfed0
+MxDSFile::~MxDSFile()
+{
+  Close();
+}
+
+// OFFSET: LEGO1 0x100cc590
+long MxDSFile::Open(unsigned long uStyle)
+{
+  // No idea what's stopping this one matching, but I'm pretty
+  // confident it has the correct behavior.
+  memset(&m_io, 0, sizeof(MXIOINFO));
+  if (m_io.Open(m_filename.Data(), uStyle) != 0) {
+    return -1;
+  }
+
+  m_io.SetBuffer(NULL, 0);
+  m_position = 0;
+
+  long longResult = 1;
+  if (m_skipReadingChunks == 0)
+  {
+    longResult = ReadChunks();
+  }
+  if (longResult != 0)
+  {
+    Close(); // vtable + 0x18
+    return longResult;
+  }
+  Seek(0, 0); // vtable + 0x24
+  return 0;
+}
+
+// OFFSET: LEGO1 0x100cc780
+long MxDSFile::Read(unsigned char *pch, unsigned long cch)
+{
+  if (m_io.Read((char*)pch, cch) != cch)
+    return -1;
+
+  m_position += cch;
+  return 0;
+}
+
+// OFFSET: LEGO1 0x100cc620
+long MxDSFile::ReadChunks()
+{
+  _MMCKINFO topChunk;
+  _MMCKINFO childChunk;
+  char tempBuffer[80];
+  
+  topChunk.fccType = 0x494e4d4f;
```

Comment:
Might be good to add a comment mentioning what this is (it's `OMNI` in a little-endian integer)